src/test/scala/is/hail/expr/ir/ArrayFunctionsSuite.scala

```diff
@@ -0,0 +1,251 @@
+package is.hail.expr.ir
+
+import is.hail.expr.types._
+import is.hail.TestUtils._
+import is.hail.expr.ir.TestUtils._
+import org.testng.annotations.{DataProvider, Test}
+import org.scalatest.testng.TestNGSuite
+
+class ArrayFunctionsSuite extends TestNGSuite {
+  val naa = NA(TArray(TInt32()))
+
+  @DataProvider(name = "basic")
```

Comment:
This is a feature of TestNG that I think is a nice way to keep IR tests compact and readable. The DataProvider is an `Array[Array[Any]]`, and a test method using that DataProvider gets run once per element of the outer array, with its parameters filled by the contents of the inner array. These are run as completely independent tests, so the report tells you exactly which inputs passed/failed.

Reply:
cool!